qiskit/pulse/library/symbolic_pulses.py

```diff
@@ -373,16 +377,6 @@ def __init__(
         if parameters is None:
             parameters = {}
 
-        if "amp" not in parameters:
-            # See https://github.com/symengine/symengine.py/issues/406 for details.
-            raise PulseError(
-                "SymbolicPulse assumes the function form 'amp' * F(t, 'duration', 'parameters'). "
-                "You must provide 'amp' in the the 'parameters' dictionary. "
-                "Because of the limitation in the symbolic computation library, "
-                "only this parameter can take a complex value for now. "
-                "This limitation might be removed in future."
-            )
-        self._amp = parameters.pop("amp")
         self._pulse_type = pulse_type
         self._params = parameters
```

Comment:
Yes, but treating `amp` as special parameter is a convention in IBM backend and this code is backend-agnostic. So one might want treat `amp` as a real value in phasor representation (I really prefer this).

Reply:
Hmm, I agree with the spirit of not treating amp, but we are still treating it specially in all the subclasses and in the parameter assignment. I think it would be more consistent just to cast it to complex here for now.

I'd like to look at what it would take to get the backend to accept float or complex for all the parameters so terra does not have to special case parameters, but that would be follow up work.

I also prefer the phasor representation. I think one can use `a * exp(1j * phi)` for the `amp` paremeter where `a` and `phi` are paraemters  to get this?

Reply:
Fair enough. This is forcibly typecasted in parameter assignment anyways. Done in https://github.com/Qiskit/qiskit-terra/pull/7821/commits/ef35cd9da61921bc41c5aa929eb1d04b1844ef3b. It would be great if we can remove this check since current implementation means we cannot write pulses with phase representation. 

> I think one can use a * exp(1j * phi) for the amp paremeter where a and phi are paraemters to get this?

Yes, this is a technique I often use, i.e.

```python
amp = Parameter("amp")
phase = Parameter("phase")
duration = Parameter("duration")
sigma = Parameter("sigma")

Gaussian(duration=duration, amp=amp * exp(1j * phase), sigma=sigma)
```